File: get_stack.js
const { execSync } = require('child_process');

const logs = execSync('docker logs garage_backend --since 5s 2>&1', { encoding: 'utf-8' });
const lines = logs.split('\n');
for (const line of lines) {
  try {
    const entry = JSON.parse(line.trim());
    if (entry.message && entry.message.includes('Unhandled error')) {
      console.log(JSON.stringify(entry, null, 2));
      break;
    }
  } catch (e) {}
}

File: get_stack.py
import sys, json

data = sys.stdin.read()
lines = data.split('\n')
for line in lines:
    line = line.strip()
    if not line:
        continue
    try:
        entry = json.loads(line)
        if entry.get('requestId') == 'req_1782096698811_61099834' and 'Unhandled error' in entry.get('message', ''):
            print(json.dumps(entry, indent=2))
            break
    except:
        pass
